beetsplug/rewrite.py

```diff
@@ -34,9 +34,7 @@ def rewriter(field, rules):
     def fieldfunc(item):
         value = item._values_fixed[field]
         for pattern, replacement in rules:
-            if pattern.match(value.lower()):
-                # Rewrite activated.
-                return replacement
+            return re.sub(pattern, replacement, value)
```

Comment:
I also just noticed this will cause another behavior change. Currently, if the pattern is triggered, the whole string is replaced with the new value. This changes the plugin to replace substrings instead.

Maybe we need a few tests to show that the old behavior still works?